File: 03-1-login_kt2/app/src/main/java/com/example/login_kt2/repository/room/LoginTry.kt
package com.example.login_kt2.repository.room

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class LoginTry(
    @PrimaryKey(autoGenerate = true)
    val id : Int = 0,
    val username : String,
    val passwordLength : Int,
    val wasSuccessful : Boolean,
    val whenTried : String
)

File: flutter_projects/03-1-login_fl/lib/repository/floor/login_try.dart

import 'package:floor/floor.dart';

@entity
class LoginTry {
  @PrimaryKey(autoGenerate: true)
  int? id;

  final String username;

  final int passwordLength;

  final bool wasSuccessful;

  final String whenTried;

  LoginTry({this.id,
    required this.username,
    required this.passwordLength,
    required this.wasSuccessful,
    required this.whenTried});

  @override
  String toString() {
    return 'LoginTry(id: $id, username: $username, passwordLength: $passwordLength, wasSuccessful: $wasSuccessful, whenTried: $whenTried)';
  }
}